pipeline/authn/authenticator_cookie_session_test.go

```diff
@@ -249,6 +250,56 @@ func TestAuthenticatorCookieSession(t *testing.T) {
 	})
 }
 
+func TestPrepareRequest(t *testing.T) {
+	t.Run("prepare request should return only configured headers", func(t *testing.T) {
+		testCases := []struct {
+			requestHeaders  []string
+			expectedHeaders []string
+			conf            *AuthenticatorCookieSessionConfiguration
+		}{
+			{
+				requestHeaders:  []string{header.Authorization, header.AcceptEncoding},
+				expectedHeaders: []string{},
+				conf:            &AuthenticatorCookieSessionConfiguration{},
+			},
+			{
+				requestHeaders:  []string{header.Authorization, header.AcceptEncoding},
+				expectedHeaders: []string{header.AcceptEncoding},
+				conf: &AuthenticatorCookieSessionConfiguration{
+					// This value is coming from the configuration and may use incorrect casing.
+					ForwardHTTPHeaders: []string{
+						"acCept-enCodinG",
```

Comment:
I added this test to show that the header config needs to match the canonical values exactly or it won't work. Can you please fix that?

Reply:
Fixed it